command/agent/alloc_endpoint.go

```diff
@@ -105,43 +106,70 @@ func (s *HTTPServer) ClientAllocRequest(resp http.ResponseWriter, req *http.Requ
 }
 
 func (s *HTTPServer) ClientGCRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
-	if s.agent.client == nil {
```

Comment:
Is there a reason this won't be nil?

Reply:
Yeah if you run just the server the client will be nil. So a config like:
```
server {
  enabled = true
  ...
}

client { // Or omitted 
  enabled = false
}
```

The reason it can be removed now is that the server can handle this endpoint now. Before if there wasn't a local client you couldn't